drivers/hmis/app/views/hmis_client/services/_table.haml

```diff
@@ -0,0 +1,27 @@
+.table-responsive
+  %table.table
+    %thead
+      %tr
+        %th
+        %th Project
+        %th Entry Date
+        %th Service Date Provided
+        %th Service Name
+        %th
+    %tbody
+      - custom_services.each do |service|
+        %tr
+          %td
+            .ds{class: "ds-color-#{service.data_source_id}"}= ds_short_name_for(service.client.id)
+          %td= service.project.name
```

Comment:
I think this needs to be `service.project.as_warehouse.name(current_user)` to respect confidential projects, right @eanders? That might also benefit from some caching to store a map of projects, though, since `as_warehouse` does a lookup currently.

Reply:
That sounds correct to me, and also too bad that it will cause N+1s.  I guess we could cache the answer to `as_warehouse` so at least we would only have to pull it once per project.

Reply:
Something along these lines?